### DISCUSSION TITLE
UseNLS switch not working in Linux Docker Containers for Dot net Core 6 Application

### DISCUSSION THREAD

**Discussion by JohnVarkey** [dotnet/runtime#89562] [Q&A] [answered]
I notice difference in the CultureInfo when running in Windows and Docker Container. On digging through I found out NLS and ICU. Does using NLS in linux will guarantee same behaviour in both Windows and Linux?

This is the piece of code that I found in the Documentation to check whether Application is using ICU 
`SortVersion sortVersion = CultureInfo.InvariantCulture.CompareInfo.Version;
byte[] bytes = sortVersion.SortId.ToByteArray();
int version = bytes[3] << 24 | bytes[2] << 16 | bytes[1] << 8 | bytes[0];
Console.WriteLine(version != 0 && version == sortVersion.FullVersion);`

I have put UseNLS as true in the csproj file. But the above code prints True for Docker container and False for Windows

`<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net6.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <DockerDefaultTargetOS>Linux</DockerDefaultTargetOS>
    <DockerfileContext>.</DockerfileContext>
	<GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
  </PropertyGroup>

  <ItemGroup>
	 <RuntimeHostConfigurationOption Include="System.Globalization.UseNls" Value="true" />
  </ItemGroup>
</Project>`

I am installing the required Locale in the Dockerfile 

`#See https://aka.ms/containerfastmode to understand how Visual Studio uses this Dockerfile to build your images for faster debugging.

FROM mcr.microsoft.com/dotnet/runtime:6.0 AS base
WORKDIR /app
RUN apt-get update \
    && apt-get install -y --allow-unauthenticated locales

#Installing en_US locale
RUN sed -i 's/^# *\(en_US.UTF-8\)/\1/' /etc/locale.gen && locale-gen

FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build
WORKDIR /src
COPY ["LocaleIssue.csproj", "."]
RUN dotnet restore "./LocaleIssue.csproj"
COPY . .
WORKDIR "/src/."
RUN dotnet build "LocaleIssue.csproj" -c Release -o /app/build

FROM build AS publish
RUN dotnet publish "LocaleIssue.csproj" -c Release -o /app/publish /p:UseAppHost=false

FROM base AS final
WORKDIR /app
ENV LANG en_US.UTF-8
ENV LC_ALL en_US.UTF-8
ENV DOTNET_SYSTEM_GLOBALIZATION_USENLS=true
COPY --from=publish /app/publish .
ENTRYPOINT ["dotnet", "LocaleIssue.dll"]`

Even after doing all of this the I still can't get the same Culture behaviour in Linux Containers and Windows. I would like to get some insights since most documentation do talk about using NLS in windows or using ICU in windows but mostly nothing about other OSs

**Comment by MichalPetryka** [comment, marked as answer]
NLS can be only used on Windows. ICU has been the default for Windows on versions where it's available for some time already.
